Add box-reflect polyfill for browsers not supporting -webkit-box-reflect (#20404)
mdn/content: files/en-us/web/css/element/index.md

{{EmbedLiveSample("Page_Preview")}}

> ### Emulating -webkit-box-reflect

Author: You're probably right. It should be on -webkit-box-reflect.

> Firefox does not support -webkit-box-reflect. You can emulate it -to some extent- using element. 

Reviewer: This needs to be elaborated on. using element? not sure what you are trying to say.

Author: yes I meant -moz-element.

Reviewer 2: ```suggestion
In browsers not supporting {{cssxref("-webkit-box-reflect")}}, can achieve the same outcome using `element`. 
```

#### HTML for webkit

```html

> <img id="pictureWReflection"

Reviewer: all images need to include alt text.

Author: I'm using it in a web app so no need for an alt.

Reviewer: MDN is for all developers. it's not about **your** web app.

Author: I wanted to keep the exemple as minimal as possible to show what's important.

Reviewer 2: ```suggestion
<img alt="A reflected image of …" id="pictureWReflection"
```

Author: Alt is not mandatory. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img
So add it if you want. But I won't.

>      src="/media/cc0-images/grapefruit-slice-332-332.jpg">

Reviewer 2: As this is not a live example and this image not stored here, we should use a placeholder name:
```suggestion
     src="pictureToReflect.jpg">
```

```

#### CSS for webkit

```css
#pictureWReflection {

>     -webkit-box-reflect: below 20px -webkit-gradient(linear, left bottom, left top, from(rgba(255,255,255,1)), color-stop(40%, transparent) , to(transparent));

Reviewer: Should use un-prefixed gradient syntax. See https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient. If you are trying to target only browsers that support -webkit-box-reflect, use @supports. 

```css
@supports (-webkit-box-reflect) {
   #pictureWReflection {
     /* valid, modern code here. no vendor prefix */
   }
}
```

Author: I don't know if un-prefixed syntax will wok.

}
```

#### HTML for Firefox

```html

> <span id="#container">

Reviewer: container is a very generic term. maybe `reflectionContainer` or some indicator of the function. Also, better to use a class name than an id so the reflection is reusable, and then maybe the id for just the `background-image` css.

Author: I disagree. The webkit version is a no-brainer where you could use class for every instance. Firefox workaround will need an id for every image that will need reflection. I don't see any interest in mixing class for the container and id for the element. Just be straight/honest to the people and tell them right away firefox version is inferior because it needs an id per image.

Unless someone can think of a better implementation that does not need it.

<img id="pictureWReflection"
     src="/media/cc0-images/grapefruit-slice-332-332.jpg">
</span>
```

#### CSS for firefox

```css
#container {
 position:relative;
}
#container:after
{
 background: -moz-element(#pictureWReflection) no-repeat;

>  transform:scaleY(-1);

Reviewer: include a space betwee the `:` and the value.
this code needs to be commented or explained.
